### PR TITLE
gh-99770: Make the correct call specialization fail kind show up

### PR DESCRIPTION
**Pull request by penguin-wwy**
* Issue: gh-99770

### CODE REVIEW

**Comment by markshannon** [comment]
@penguin-wwy 

Sorry, this got overlooked when I was updating the stats for `LOAD_ATTR` and `CALL`.

We still don't specify whether the failure is for a builtin function or method descriptor, so this PR would still be a valuable addition, if you are willing to update it.

**Comment by penguin-wwy** [comment]
I updated the failure for the method descriptor.

File: Python/specialize.c [hunk 1]
```diff
@@ -1251,10 +1251,10 @@ static int
 function_kind(PyCodeObject *code) {
     int flags = code->co_flags;
     if ((flags & (CO_VARKEYWORDS | CO_VARARGS)) || code->co_kwonlyargcount) {
-        return SPEC_FAIL_CALL_COMPLEX_PARAMETERS;
+        return SPEC_FAIL_CODE_COMPLEX_PARAMETERS;
```
Comment (penguin-wwy) [inline Python/specialize.c:1254]: `function_kind` used in `subscr`、`attr` and `call` specialize, move the failure to the common category.

File: Python/specialize.c [hunk 2]
```diff
@@ -309,26 +309,27 @@ _PyCode_Quicken(PyCodeObject *code)
 #define SPEC_FAIL_OUT_OF_RANGE 4
 #define SPEC_FAIL_EXPECTED_ERROR 5
 #define SPEC_FAIL_WRONG_NUMBER_ARGUMENTS 6
-#define SPEC_FAIL_NOT_PY_FUNCTION 7
+#define SPEC_FAIL_CODE_COMPLEX_PARAMETERS 7
+#define SPEC_FAIL_CODE_NOT_OPTIMIZED 8
```
Comment (markshannon) [inline Python/specialize.c:313]: You will need to update the summarize_stats script
https://github.com/python/cpython/blob/main/Tools/scripts/summarize_stats.py#L227

Comment (markshannon) [inline Python/specialize.c:313]: You will also need to change `SPEC_FAIL_UNPACK_SEQUENCE_ITERATOR`

**Review by markshannon — CHANGES_REQUESTED** [review]
Thanks for doing this. There a couple more values that need changing.

File: Python/specialize.c (hunk 2 above)
Comment (penguin-wwy) [reply Python/specialize.c:313]: Done

Comment (penguin-wwy) [reply Python/specialize.c:313]: Done

**Comment by penguin-wwy** [comment]
I have made the requested changes; please review again.

**Review by markshannon — APPROVED** [review]
That's great. Thanks.

**Pull request merged**